src/WebJobs.Script.Grpc/Extensions/ScriptInvocationContextExtensions.cs

```diff
@@ -33,9 +35,16 @@ public static async Task<InvocationRequest> ToRpcInvocationRequest(this ScriptIn
                 TypedData rpcValue = null;
                 if (input.val == null || !rpcValueCache.TryGetValue(input.val, out rpcValue))
                 {
-                    rpcValue = await input.val.ToRpc(logger, capabilities);
+                    // TODO gochaudh: Check config before using SharedMemoryManager
+                    rpcValue = await input.val.ToRpc(logger, capabilities, sharedMemoryManager);
```

Comment:
- check `capabilities` if shared memory data feature is supported
   - If yes, enforce minimum size of the payload - 1MB?
       -  if size > 1MB , create memory map
        - if creating memory map fails - fall back to existing code path

Reply:
Also, add check for environment variable - `Functions_Worker_SharedMemoryData` is not `false` -- this is to ensure users have ability to turn off the feature via AppSettings.